server/src/main/java/org/candlepin/resource/util/ResolverUtil.java

```diff
@@ -78,35 +80,57 @@ public Owner resolveOwner(Owner owner) {
         return owner;
     }
 
-    public Product resolveProduct(Owner owner, Product product) {
-        String id = null;
-        if (product != null) {
-            id = product.getId();
-        }
-        return resolveProduct(owner, id);
+    public Product resolveProduct(Owner owner, String productId, boolean allowNull) {
+        Product product = new Product();
+        product.setId(productId);
+        return resolveProduct(owner, product, allowNull);
     }
 
-    public Product resolveProduct(Owner owner, String productId) {
-        if (productId == null) {
-            throw new BadRequestException(
-                    i18n.tr("No product specified, or product lacks identifying information"));
+    public ProductData resolveProduct(Owner owner, ProductData product, boolean allowNull) {
+        Product argProduct = new Product();
+        ProductData result = null;
+        if (product != null) {
+            argProduct.setId(product.getId());
+            argProduct.setUuid(product.getUuid());
         }
-
-        // TODO: Maybe add UUID resolution as well?
-        return findProduct(owner, productId);
+        if (product != null) {
```

Comment:
Duplicate null check.

Also, include whitespace between your blocks like this. With our style forcing newlines between blocks in an if-elseif-else, the lack of whitespace here makes them look connected at a glance.

```allowNull``` isn't working properly here. If you set ```allowNull``` to false and then provide a null DTO, the call to resolveProduct that throws the exception is never called.